src/openzaak/components/besluiten/models.py

```diff
@@ -43,42 +45,73 @@ class Besluit(AuditTrailMixin, APIMixin, models.Model):
         "organisatie die het besluit heeft vastgesteld.",
         db_index=True,
     )
-
-    _besluittype_url = models.URLField(
-        _("extern besluittype"),
+    _besluittype_base_url = models.ForeignKey(
+        Service,
+        null=True,
         blank=True,
-        max_length=1000,
-        help_text=_(
-            "URL-referentie naar extern BESLUITTYPE (in een andere Catalogi API)."
-        ),
+        on_delete=models.PROTECT,
+        related_name="besluit_besluittypen",
+        help_text="Basis deel van URL-referentie naar het extern BESLUITTYPE (in een andere Catalogi API).",
+    )
+    _besluittype_relative_url = models.CharField(
+        _("besluittype relative url"),
+        max_length=200,
+        blank=True,
+        null=True,
+        help_text="Relatief deel van URL-referentie naar het extern BESLUITTYPE (in een andere Catalogi API).",
+    )
+    _besluittype_url = ServiceUrlField(
+        base_field="_besluittype_base_url",
+        relative_field="_besluittype_relative_url",
+        blank=True,
+        null=True,
+        help_text="URL-referentie naar het extern BESLUITTYPE (in een andere Catalogi API).",
     )
```

Comment:
I can't tell easily from the migration files - does this field also set up a check constraint that when the FK is filled out, there must be a non-empty value for the `CharField` too?

Reply:
the check constraint added to `zgw_consumers.ServiceUrlField` (PR - https://github.com/maykinmedia/zgw-consumers/pull/56)